From ea78afec7d9c5f327e2aef37e619323ff17b10eb Mon Sep 17 00:00:00 2001
From: liwei <liwei728@huawei.com>
Date: Thu, 23 May 2024 21:46:09 +0800
Subject: [PATCH 755/960] downstream: Revert "downstream: ACPI: Rename
 ACPI_HOTPLUG_CPU to include 'present'"

This reverts commit 4eea83320031.

To keep CONFIG unchanged, modify ACPI_HOTPLUG_PRESENT_CPU to
ACPI_HOTPLUG_CPU.

Conflicts:
	arch/x86/Kconfig
	drivers/acpi/acpi_processor.c
	include/linux/acpi.h
Signed-off-by: liwei <liwei728@huawei.com>
Signed-off-by: nzq-hw <nizhiqiang1@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/x86/Kconfig              | 2 +-
 arch/x86/kernel/acpi/boot.c   | 4 ++--
 drivers/acpi/Kconfig          | 6 +++---
 drivers/acpi/acpi_processor.c | 4 ++--
 drivers/acpi/bus.c            | 2 +-
 include/linux/acpi.h          | 2 +-
 6 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 130a8bbfa805..9b6cd8bddaa6 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -61,7 +61,7 @@ config X86
 	#
 	select ACPI_LEGACY_TABLES_LOOKUP	if ACPI
 	select ACPI_SYSTEM_POWER_STATES_SUPPORT	if ACPI
-	select ACPI_HOTPLUG_PRESENT_CPU		if ACPI_PROCESSOR && HOTPLUG_CPU
+	select ACPI_HOTPLUG_CPU			if ACPI_PROCESSOR && HOTPLUG_CPU
 	select ACPI_HOTPLUG_IGNORE_OSC		if ACPI && HOTPLUG_CPU
 	select ARCH_32BIT_OFF_T			if X86_32
 	select ARCH_CLOCKSOURCE_INIT
diff --git a/arch/x86/kernel/acpi/boot.c b/arch/x86/kernel/acpi/boot.c
index 937b4e77bc9f..fbba2fa62e75 100644
--- a/arch/x86/kernel/acpi/boot.c
+++ b/arch/x86/kernel/acpi/boot.c
@@ -734,7 +734,7 @@ static void __init acpi_set_irq_model_ioapic(void)
 /*
  *  ACPI based hotplug support for CPU
  */
-#ifdef CONFIG_ACPI_HOTPLUG_PRESENT_CPU
+#ifdef CONFIG_ACPI_HOTPLUG_CPU
 #include <acpi/processor.h>
 
 static int acpi_map_cpu2node(acpi_handle handle, int cpu, int physid)
@@ -777,7 +777,7 @@ int acpi_unmap_cpu(int cpu)
 	return 0;
 }
 EXPORT_SYMBOL(acpi_unmap_cpu);
-#endif				/* CONFIG_ACPI_HOTPLUG_PRESENT_CPU */
+#endif				/* CONFIG_ACPI_HOTPLUG_CPU */
 
 int acpi_register_ioapic(acpi_handle handle, u64 phys_addr, u32 gsi_base)
 {
diff --git a/drivers/acpi/Kconfig b/drivers/acpi/Kconfig
index b6ceeaf63057..e66874d4c3e3 100644
--- a/drivers/acpi/Kconfig
+++ b/drivers/acpi/Kconfig
@@ -305,14 +305,14 @@ config ACPI_IPMI
 	  To compile this driver as a module, choose M here:
 	  the module will be called as acpi_ipmi.
 
-config ACPI_HOTPLUG_PRESENT_CPU
+config ACPI_HOTPLUG_CPU
 	bool
 	depends on ACPI_PROCESSOR && HOTPLUG_CPU
 	select ACPI_CONTAINER
 
 config ACPI_HOTPLUG_IGNORE_OSC
 	bool
-	depends on ACPI_HOTPLUG_PRESENT_CPU
+	depends on ACPI_HOTPLUG_CPU
 	help
 	  Ignore whether firmware acknowledged support for toggling the CPU
 	  present bit in _STA. Some architectures predate the _OSC bits, so
@@ -408,7 +408,7 @@ config ACPI_PCI_SLOT
 
 config ACPI_CONTAINER
 	bool "Container and Module Devices"
-	default (ACPI_HOTPLUG_MEMORY || ACPI_HOTPLUG_PRESENT_CPU)
+	default (ACPI_HOTPLUG_MEMORY || ACPI_HOTPLUG_CPU)
 	help
 	  This driver supports ACPI Container and Module devices (IDs
 	  ACPI0004, PNP0A05, and PNP0A06).
diff --git a/drivers/acpi/acpi_processor.c b/drivers/acpi/acpi_processor.c
index 2aae3b9b8794..55f831e15cb5 100644
--- a/drivers/acpi/acpi_processor.c
+++ b/drivers/acpi/acpi_processor.c
@@ -184,7 +184,7 @@ static void __init acpi_pcc_cpufreq_init(void) {}
 
 static bool acpi_processor_hotplug_present_supported(void)
 {
-	if (!IS_ENABLED(CONFIG_ACPI_HOTPLUG_PRESENT_CPU))
+	if (!IS_ENABLED(CONFIG_ACPI_HOTPLUG_CPU))
 		return false;
 
 	/* x86 systems pre-date the _OSC bit */
@@ -503,7 +503,7 @@ static void acpi_processor_make_not_present(struct acpi_device *device)
 {
 	struct acpi_processor *pr;
 
-	if (!IS_ENABLED(CONFIG_ACPI_HOTPLUG_PRESENT_CPU)) {
+	if (!IS_ENABLED(CONFIG_ACPI_HOTPLUG_CPU)) {
 		pr_err_once("Changing CPU present bit is not supported");
 		return;
 	}
diff --git a/drivers/acpi/bus.c b/drivers/acpi/bus.c
index 54af369bee16..a5e37b5e2581 100644
--- a/drivers/acpi/bus.c
+++ b/drivers/acpi/bus.c
@@ -355,7 +355,7 @@ static void acpi_bus_osc_negotiate_platform_control(void)
 		capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_APEI_SUPPORT;
 
 	capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_HOTPLUG_ENABLED_SUPPORT;
-	if (IS_ENABLED(CONFIG_ACPI_HOTPLUG_PRESENT_CPU))
+	if (IS_ENABLED(CONFIG_ACPI_HOTPLUG_CPU))
 		capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_HOTPLUG_PRESENT_SUPPORT;
 
 	if (ACPI_FAILURE(acpi_get_handle(NULL, "\\_SB", &handle)))
diff --git a/include/linux/acpi.h b/include/linux/acpi.h
index 6a11bec09702..4bed59ed3ea0 100644
--- a/include/linux/acpi.h
+++ b/include/linux/acpi.h
@@ -651,7 +651,7 @@ static inline u32 acpi_osc_ctx_get_cxl_control(struct acpi_osc_context *context)
 #define ACPI_GSB_ACCESS_ATTRIB_RAW_PROCESS	0x0000000F
 
 /* Enable _OST when all relevant hotplug operations are enabled */
-#if defined(CONFIG_ACPI_HOTPLUG_PRESENT_CPU) &&			\
+#if defined(CONFIG_ACPI_HOTPLUG_CPU) &&			\
 	defined(CONFIG_ACPI_HOTPLUG_MEMORY) &&		\
 	defined(CONFIG_ACPI_CONTAINER)
 #define ACPI_HOTPLUG_OST
-- 
2.43.0

